src/Npgsql/SqlQueryParser.cs

```diff
@@ -300,7 +301,7 @@ internal void ParseRawQuery(string sql, bool standardConformantStrings, NpgsqlPa
             goto Finish;
 
         DollarQuoted: {
-                var tag = sql.Substring(dollarTagStart - 1, dollarTagEnd - dollarTagStart + 2);
+                var tag = Substring(sqlSpan, dollarTagStart - 1, dollarTagEnd - dollarTagStart + 2).ToString();
                 var pos = sql.IndexOf(tag, dollarTagEnd + 1); // Not linear time complexity, but that's probably not a problem, since PostgreSQL backend's isn't either
```

Comment:
Should the remaining references to `sql` be changed to `sqlSpan`? 

Of course, that would only make sense if calls/indexers on `ReadOnlySpan<char>` are _at least not slower_ than the equivalent calls against `string`. I'm not sure of the specifics, but would you be able to look into it?

Reply:
I'll have a look. To my knowledge at worst a call to Span use the underlying string method so it shouldn't be too bad.